src/sentry/static/sentry/app/views/organizationDashboard/exploreWidget.jsx

```diff
@@ -1,69 +1,119 @@
+  getExportToDiscover = query => {
+    const {organization} = this.props;
     const {
       datetime,
       environments, // eslint-disable-line no-unused-vars
       ...selection
     } = this.props.selection;
 
-    event.stopPropagation();
-
     // Discover does not support importing these
     const {
       groupby, // eslint-disable-line no-unused-vars
       rollup, // eslint-disable-line no-unused-vars
       orderby,
-      ...query
-    } = firstQuery;
+      ...restQuery
+    } = query;
 
     const orderbyTimeIndex = orderby.indexOf('time');
     let visual = 'table';
 
     if (orderbyTimeIndex !== -1) {
-      query.orderby = `${orderbyTimeIndex === 0 ? '' : '-'}${query.aggregations[0][2]}`;
+      restQuery.orderby = `${orderbyTimeIndex === 0 ? '' : '-'}${restQuery
+        .aggregations[0][2]}`;
       visual = 'line-by-day';
     } else {
-      query.orderby = orderby;
+      restQuery.orderby = orderby;
     }
 
-    router.push(
-      `/organizations/${organization.slug}/discover/${getQueryStringFromQuery({
-        ...query,
-        ...selection,
-        start: datetime.start,
-        end: datetime.end,
-        range: datetime.period,
-        limit: 1000,
-      })}&visual=${visual}`
-    );
+    return `/organizations/${organization.slug}/discover/${getQueryStringFromQuery({
+      ...restQuery,
+      ...selection,
+      start: datetime.start,
+      end: datetime.end,
+      range: datetime.period,
+      limit: 1000,
+    })}&visual=${visual}`;
+  };
+
+  getExportToEvents = query => {
+    const {selection, organization} = this.props;
+    return getEventsUrlPathFromDiscoverQuery({
+      organization,
+      selection,
+      query,
+    });
   };
 
   render() {
-    // TODO(billy): This is temporary
-    // Need design followups
+    const {widget} = this.props;
+    const discoverQueries = widget.queries.discover;
+
     return (
-      <Button size="xsmall" onClick={this.handleExportToDiscover}>
-        <InlineSvg src="icon-discover" />
-      </Button>
+      <DropdownLink
+        anchorRight={true}
+        title={t('Explore')}
+        topLevelClasses={exploreMenuCss}
+      >
+        {discoverQueries.map(query => (
+          <ExploreRow key={query.name}>
+            <QueryName>{query.name}</QueryName>
+
+            <ExploreActions>
+              <Button borderless size="zero" to={this.getExportToDiscover(query)}>
+                <InlineSvg src="icon-discover" />
+              </Button>
+              <Button borderless size="zero" to={this.getExportToEvents(query)}>
+                <InlineSvg src="icon-stack" />
+              </Button>
+            </ExploreActions>
+          </ExploreRow>
+        ))}
+      </DropdownLink>
     );
   }
 }
 export default withOrganization(ExploreWidget);
+
+const ExploreRow = styled('li')`
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  padding: 0 ${space(0.5)};
+`;
+
+const ExploreActions = styled('div')`
```

Comment:
if you add `flex-grow: 1` to `QueryName`, then I don't think you need this div or the `justify-content` declaration in `ExploreRow`

Reply:
I'll address this in the follow up PR.